protocols/gossipsub/src/behaviour.rs

```diff
@@ -921,16 +963,63 @@ impl Gossipsub {
 
             for peer in recipient_peers.iter() {
                 debug!("Sending message: {:?} to peer {:?}", msg_id, peer);
-                self.events.push_back(NetworkBehaviourAction::NotifyHandler {
-                    peer_id: peer.clone(),
-                    event: event.clone(),
-                    handler: NotifyHandler::Any,
-                });
+                self.events
+                    .push_back(NetworkBehaviourAction::NotifyHandler {
+                        peer_id: peer.clone(),
+                        event: event.clone(),
+                        handler: NotifyHandler::Any,
+                    });
             }
         }
         debug!("Completed forwarding message");
     }
 
+    /// Constructs a `GossipsubMessage` performing message signing if required.
+    pub(crate) fn build_message(
+        &self,
+        topics: Vec<TopicHash>,
+        data: Vec<u8>,
+    ) -> Result<GossipsubMessage, String> {
```

Comment:
```suggestion
    ) -> Result<GossipsubMessage, SigningError> {
```

While not opinionated on which error handling pattern to use I would prefer not to pass `String`s around as one can not programmatically react on them.